### DISCUSSION TITLE
Support MySQL Driver

### DISCUSSION THREAD

**Discussion by marvin-wtt** [thedevs-network/kutt#761] [Ideas]
I'd like to add support for MySQL and possibly other DB drivers too. 

Do do so, all raw queries need to be removed. The uuids are very simple as knex offers a uuid utility function since the last release. 

The `cooldowns` column of the `users` table is a bit more complicated, as arrays are not supposed by MySQL. I'd either replace it by a JSON object or preferably just drop the timestamps and replace it with an incremental integer. Any thoughts?

Also, should I base the PR on develop or the V3 branch?

**Comment by poeti8** [comment]
Hey, 

Yes I had seen the knex's uuid method and I've already updated the knex package along with others in the v3 branch, so it should be ready to use there.

I like the idea of integers. We can have one `malware_attempts` integer field for the number of bad attempts, and one `cooldown` timestamp field to store the date. 

Both Postgres and MySQL support the small integer data type, we can use that for the integer field.

If you to work on it use the v3 branch please. Thanks.

**Reply by marvin-wtt** [reply]
Ok, thanks for the fast response.

Just out of curiosity, is there any particular reason why you dropped typescript?

**Reply by poeti8** [reply]
Yes, the reason is I couldn't find an enough convincing reason to keep using it.

**Reply by marvin-wtt** [reply]
IMO TS just helps to prevent errors but of course that's just my opinion
